RELEASING.md

```diff
@@ -0,0 +1,26 @@
+# Making a Release
+
+- Ensure the referenced specification version is up to date. Use
+  [tooling to update the spec](./CONTRIBUTING.md#updating-the-referenced-specification-version)
+  if needed.
+- Run [opentelemetry.io workflow](https://github.com/open-telemetry/opentelemetry.io/actions/workflows/build-dev.yml)
+  against `semantic-conventions` submodule as a smoke-test for docs. Fix broken links, if any.
```

Comment:
we don't need it anymore, right?

```suggestion
```

Reply:
it seems we do, this is how I found #1766 (it was an [intra-spec link to a page title](https://github.com/open-telemetry/opentelemetry-specification/pull/4345#issuecomment-2573599798))